### ISSUE TITLE
Annotations work only if global aes() have been defined

### ISSUE THREAD

**Issue by bleutner** [hadley/ggplot2#1655]
Hi, 

adding an annotation works only if global aesthetics have been defined in `ggplot()`.
If there are only "local" aesthetics in individual geoms the annotation will not be drawn (no warning though).

```r
library(ggplot2)
library(grid)
df <- data.frame(x = 1:10, y= 1:10)	
anno <- annotation_custom(grid.rect(draw=FALSE), xmin = 7, xmax = 10,ymin = 1, ymax = 2)

## Global aesthetics: works fine
ggplot(df, aes(x = x, y = y)) + geom_point()+ 
		anno

## "Local" aesthetics: does not work
ggplot() + geom_point(data = df, aes(x = x, y = y))+
		anno 
```
![selection_065](https://cloud.githubusercontent.com/assets/1650314/16339190/a41b8ac2-3a21-11e6-8272-550a8b17697e.png)

![selection_066](https://cloud.githubusercontent.com/assets/1650314/16339193/aaa60160-3a21-11e6-8cdb-df9018bcb251.png)

I tried setting `scales` and/or `coord` limits explicitly, but this doesn't make it work either:
		
```r
ggplot() + geom_point(data = df, aes(x = x, y = y))+
		anno +
		scale_x_continuous(limits = c(-10,10)) +
		scale_y_continuous(limits = c(-10,10)) +
		coord_equal(xlim = c(-20,20), ylim = c(-20,20))
```

**Comment by hadley** [comment]
The problem is that annotation currently sets `data = NULL` and clearly there's a bug somewhere because this causes `draw_panel()` to never be rendered.

Minimal reprex:

```R
library(ggplot2)
df <- data.frame(x = 1:10, y = 1:10) 
ggplot() + 
  geom_point(aes(x, y), data = df) + 
  annotation_custom(grid::rectGrob(), xmin = 7, xmax = 10, ymin = 1, ymax = 2)
```

**Comment by thomasp85** [comment]
This is due to Layer checking for empty data before calling `draw_layer`. I'll add a switch to Layer to allow drawing with no data

**Comment by thomasp85** [comment]
This is being fixed in #1690
